src/ExplorerController.ts

```diff
@@ -156,13 +237,12 @@ export class ExplorerController {
      * @param diag diagnostic to show
      * @param mappingUpdate optional flag to indicate a mapping update and the state shouldn't be reset
      */
-    public setActiveDiagnostic(diag: SarifViewerDiagnostic, mappingUpdate?: boolean) {
-        if (this.activeSVDiagnostic === undefined || this.activeSVDiagnostic !== diag || mappingUpdate) {
-            this.activeSVDiagnostic = diag;
+    public setActiveDiagnostic(diag: SarifViewerVsCodeDiagnostic, mappingUpdate?: boolean): void {
+        if (!this.activeDiagnostic || this.activeDiagnostic !== diag || mappingUpdate) {
+            this.activeDiagnostic = diag;
             if (!mappingUpdate) {
                 this.activeTab = undefined;
-                this.selectedRow = undefined;
-                this.selectedVerbosity = undefined;
```

Comment:
Verbosity no longer resets on mapping update?

Reply:
This is just a functional bug. The level of verbosity is not intended to change as you move between individual diagnostics.

---
In reply to: [397533596](https://github.com/microsoft/sarif-vscode-extension/pull/211#discussion_r397533596) [](ancestors = 397533596)